src/gbxml/ForwardTranslator.cpp

```diff
@@ -1350,22 +1327,39 @@ namespace gbxml {
     return result;
   }
 
+  void ForwardTranslator::translateId(const openstudio::model::ModelObject& modelObject, pugi::xml_node& parentElement) {
+
+    std::string id = modelObject.name().get();
+    if (modelObject.gbXMLId()) {
+      id = modelObject.gbXMLId().get();
+    }
+    parentElement.append_attribute("id") = escapeName(id).c_str();
+  }
+
+  void ForwardTranslator::translateName(const openstudio::model::ModelObject& modelObject, pugi::xml_node& parentElement) {
+
+    std::string name = modelObject.name().get();
+    if (modelObject.displayName()) {
+      name = modelObject.displayName().get();
+    }
+    parentElement.append_child("Name").text() = name.c_str();
+  }
+
   boost::optional<pugi::xml_node> ForwardTranslator::translateCADObjectId(const openstudio::model::ModelObject& modelObject,
                                                                           pugi::xml_node& parentElement) {
     boost::optional<pugi::xml_node> result;
 
     if (modelObject.hasAdditionalProperties()) {
       model::AdditionalProperties additionalProperties = modelObject.additionalProperties();
-      if (additionalProperties.hasFeature("CADObjectId")) {
-        boost::optional<std::string> cadObjectId = additionalProperties.getFeatureAsString("CADObjectId");
-        if (cadObjectId) {
-          if (additionalProperties.hasFeature("programIdRef")) {
-            boost::optional<std::string> programIdRef = additionalProperties.getFeatureAsString("programIdRef");
-            if (programIdRef) {
-              auto cadObjectIdElement = parentElement.append_child("CADObjectId");
-              cadObjectIdElement.append_attribute("programIdRef") = (*programIdRef).c_str();
-              result = cadObjectIdElement;
-            }
+      boost::optional<std::string> cadObjectId = modelObject.cadObjectId();
+      if (cadObjectId) {
```

Comment:
This is kinda a side note, but I would prefer always using the initialize in if construct. It limits scope, so you cannot try to access cadObjectId by mistake after the if block.

```suggestion
      if (boost::optional<std::string> cadObjectId = modelObject.cadObjectId()) {
```